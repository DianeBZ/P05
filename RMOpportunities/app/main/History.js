import React, { Component } from 'react';
import Translation from './Translation';

import {
  StyleSheet,
  Text,
  View,
  Dimensions,
  BackAndroid,
  ListView,
} from 'react-native';

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;

//We change this and add some foncs to realise 
//the display of the list of factures when we need.
var dataSource = null;

export default class History extends Component {
  render() {
	BackAndroid.addEventListener('Back',this.onBackAndroid);
    return (
      <View style={styles.container}>
        <View style={{height:height*0.08}}/>
		<View style={styles.containerMain}>
			<View style = {styles.containerText}>
				<Text style = {styles.title}>
					{Translation[lang].histo}
				</Text>
				<Text style = {styles.instructions}>
					{Translation[lang].visio_factures}
				</Text>
			</View>
			
			<View style = {styles.header}>
				<View style = {styles.listHeader}>
					<Text style = {{textAlign: 'center',}}>
						{Translation[lang].montant_com}
					</Text>
				</View>
				<View style = {styles.listHeader}>
					<Text style = {{textAlign: 'center',}}>
						{Translation[lang].date_fact}
					</Text>
				</View>
				<View style = {styles.listHeader}>
					<Text style = {{textAlign: 'center',}}>
						{Translation[lang].date_paie}
					</Text>
				</View>
				<View style = {styles.listHeader}>
					<Text style = {{textAlign: 'center',}}>
						{Translation[lang].lien}
					</Text>
				</View>
			</View>
			
			{(() => {
				if (dataSource != null)
					return (
						<ListView />
					);
				else
					return (
						<View style = {styles.containerFacture}>
							<Text>
								{Translation[lang].aucune_fact}
							</Text>
						</View>
					);
			})()}
		</View>
      </View>
    );
  }
    onBackAndroid = () => {
		const { navigator } = this.props;
		if (navigator && navigator.getCurrentRoutes().length > 1) {
			BackAndroid.removeEventListener('Back', this.onBackAndroid);
			navigator.pop();
			return true;
		} else {
			return false;
		}
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F2F2F2',
  },
  containerMain: {
    backgroundColor: '#F2F2F2',
	flexGrow: 0,
	flexShrink: 0,
	justifyContent: 'center',
	alignItems: 'center',
	flexDirection: 'column',
	width: width,
	height: height * 0.89,
	alignSelf: 'center',
  },
  containerText: {
    backgroundColor: '#F2F2F2',
	flexGrow: 0,
	flexShrink: 0,
	justifyContent: 'center',
	alignItems: 'center',
	flexDirection: 'column',
	width: width,
	height: height * 0.37,
	alignSelf: 'center',
  },
  containerFacture: {
	width: width,
	height: height * 0.38,
	flexGrow: 0,
	flexShrink: 0,
	justifyContent: 'center',
	alignItems: 'center',
	alignSelf: 'center',
	backgroundColor: '#F2F2F2',
	padding: 5,
	borderWidth: 1,  
    borderRadius: 5,
	margin: 10,
    borderColor: '#CCC',
	width: 0.92 * width + 20,
  },
  title: {
    fontSize: 40,
    textAlign: 'center',
    margin: 10,
	color: '#333333',
  },
  instructions: {
    fontSize: 15,
	textAlign: 'center',
    marginBottom: 5,
  },
  header: {
	width: width,
	height: height * 0.12,
	alignItems: 'flex-end',
	flexDirection: 'row',
	justifyContent: 'center',
  },
  listHeader: {
	justifyContent: 'center',
	alignSelf: 'center',	
    padding: 5,  
    margin: 3,  
    width: 0.23 * width,  
    height: height * 0.12, 
    alignItems: 'center',
    borderWidth: 1,  
    borderRadius: 5,  
    borderColor: '#CCC',
  },
});