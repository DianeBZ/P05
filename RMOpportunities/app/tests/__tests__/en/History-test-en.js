// __tests__/History-test-en.js
import 'react-native';
import React from 'react';

global.lang = 'en';
import History from '../../../main/History';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <History />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
