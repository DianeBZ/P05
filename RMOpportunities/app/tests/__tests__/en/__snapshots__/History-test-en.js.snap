// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F2F2F2",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "height": 106.72,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#F2F2F2",
        "flexDirection": "column",
        "flexGrow": 0,
        "flexShrink": 0,
        "height": 1187.26,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#F2F2F2",
          "flexDirection": "column",
          "flexGrow": 0,
          "flexShrink": 0,
          "height": 493.58,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 40,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        History
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 15,
            "marginBottom": 5,
            "textAlign": "center",
          }
        }
      >
        View your invoices
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "height": 160.07999999999998,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "borderColor": "#CCC",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 160.07999999999998,
            "justifyContent": "center",
            "margin": 3,
            "padding": 5,
            "width": 172.5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Commission amount
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "borderColor": "#CCC",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 160.07999999999998,
            "justifyContent": "center",
            "margin": 3,
            "padding": 5,
            "width": 172.5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Billing date
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "borderColor": "#CCC",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 160.07999999999998,
            "justifyContent": "center",
            "margin": 3,
            "padding": 5,
            "width": 172.5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Payment date
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "borderColor": "#CCC",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 160.07999999999998,
            "justifyContent": "center",
            "margin": 3,
            "padding": 5,
            "width": 172.5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Link
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#F2F2F2",
          "borderColor": "#CCC",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexGrow": 0,
          "flexShrink": 0,
          "height": 506.92,
          "justifyContent": "center",
          "margin": 10,
          "padding": 5,
          "width": 710,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        No invoice available
      </Text>
    </View>
  </View>
</View>
`;
